index.js

```diff
@@ -262,6 +265,89 @@ class ReadBatch {
   }
 }
 
+let tmpResolve = null
```

Comment:
why do these need to be top level?

Reply:
its my new pattern for avoiding that function allocation, shuold prop be a module, will make one